### ISSUE TITLE
Product/download attributes

### ISSUE THREAD

**Issue by amdrew** [easydigitaldownloads/easy-digital-downloads#6105]
The idea is to add download "attributes", "info", "meta", (whatever you want to call them) to a download. This would then be outputted via a widget, shortcode, or automatically at the bottom of a download page in some kind of list.

For example, an end-user wanting to sell eBooks with EDD could add things like the author, number of pages, ISBN number, formats available, chapters, publish date (it may differ than when the download was added) and more. A site selling photos may need to add photo dimensions, megapixels, location, camera, photographer etc.

We briefly discussed this during a meeting a few months ago. EDD has evolved to much more than selling just software and we need to cater to the various use-cases it could be used for. A nice touch could be providing pre-populated fields for specific use-cases that a customer could select from, or make it their default.

**Comment by cklosowski** [comment]
I think this is the point of the EDD Fields plugin that @brashrebel had started no?

**Comment by amdrew** [comment]
Yep, merging that into core and polishing is definitely an option. Just had a play with it now, works well.

**Comment by brashrebel** [comment]
Last time we discussed it we considered several options for that plugin:

A. Make it its own premium extension.
B. Make it a free extension.
C. Merge it into core.
D. Make something different, possibly simpler for core.

No final decision has been made. The plugin's been stable for months but we couldn't reach a consensus regarding these and even other options.

It would be good to make an official plan. Should we just discuss here? Or in Slack? Have a special meeting? I'm interested in helping get it nailed down.

**Comment by spencerfinnell** [comment]
Previously: https://github.com/easydigitaldownloads/easy-digital-downloads/issues/4301

**Comment by pippinsplugins** [comment]
I think #4301 is different.

This is referring to adding support for custom meta data that can be displayed on the frontend. #4301 is referring to more to product attributes that apply to specific price options. For example, #4301 is suggesting we provide a way to add custom product features that are tied to price options.

**Comment by amdrew** [comment]
After some discussion, D seems like the option we'll pursue.
